cpp/tests/strings/attrs_tests.cpp

```diff
@@ -93,3 +92,16 @@ TEST_F(StringsAttributesTest, StringsLengths)
     CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
   }
 }
+
+TEST_F(StringsAttributesTest, StringsLengthsLong)
+{
+  std::vector<std::string> h_strings(
+    40000, "something a bit longer than 32 bytes ééé ééé ééé ééé ééé ééé ééé");
+  cudf::test::strings_column_wrapper strings(h_strings.begin(), h_strings.end());
+  auto strings_view = cudf::strings_column_view(strings);
+
+  auto results = cudf::strings::count_characters(strings_view);
+  std::vector<int32_t> h_expected(h_strings.size(), 64);
```

Comment:
I expected that these tests should all be using `cudf::size_type` for the counts, but I also see that `count_characters` is defined to return an `INT32` in its docstring. That seems undesirable, right? Shouldn't we be using `cudf::size_type` instead of hardcoding `int32_t` for the function return value and test comparisons?

Reply:
Since there is no `SIZE_TYPE` column type, I think it is more correct to use the appropriate data type.

Reply:
If changing the definition of size type would affect the correctness or consistency of this code, we should use size type.

Reply:
I disagree but this seems out of scope for this PR.
There are at least 6 APIs that do this today.
Perhaps you can create a separate issue if you want to reopen the discussion.

We document what data-type is used for the output of the column of these APIs.
I'm not comfortable saying the column type is whatever size-type equates to at the moment.

Reply:
Discussed a bit offline. The latest documented status is https://github.com/rapidsai/cudf/issues/3958#issuecomment-1166338595 but from conversation since that was posted, @GregoryKimball seemed interested in moving towards option 3. I won't hold this PR up on that account.